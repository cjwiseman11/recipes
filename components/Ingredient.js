import React from 'react';

const Ingredient = ({ ingredient, Component = 'span', info, onHighlight, isHighlighted }) => {
  return (
    <Component
      onMouseEnter={() => onHighlight(ingredient)}
      onMouseLeave={() => onHighlight(null)}
      className={isHighlighted ? styles.hover : ''}
    >
      {ingredient}
      {info}
    </Component>
  );
};

export default Ingredient;
