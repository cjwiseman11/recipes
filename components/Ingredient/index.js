import React from 'react';
import styles from '../../styles/Home.module.css';

const Ingredient = ({
  ingredient,
  Component = 'span',
  info,
  onHighlight,
  isHighlighted,
  onClick,
}) => {
  return (
    <Component
      onMouseEnter={() => onHighlight(ingredient)}
      onMouseLeave={() => onHighlight(null)}
      onClick={onClick}
      className={`${styles.ingredient} ${isHighlighted ? styles.hover : ''}`}
    >
      {ingredient}
      {info}
    </Component>
  );
};

export default Ingredient;
