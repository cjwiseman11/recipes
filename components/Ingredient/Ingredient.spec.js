import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Ingredient from './index';

describe('<Ingredient />', () => {
  test('Ingredient is rendered with a span by default', async () => {
    const { container } = render(<Ingredient ingredient="A tasty pie" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  test('When a Component is passed, ingredient is rendered with Component', async () => {
    const component = render(<Ingredient ingredient="A tasty pie" Component="li" />);

    expect(component.getByRole('listitem')).toHaveTextContent('A tasty pie');
  });

  test('When info is passed, info text is shown', async () => {
    const component = render(
      <Ingredient ingredient="A tasty pie" Component="li" info="This has some info!" />
    );

    expect(component.getByRole('listitem')).toHaveTextContent('This has some info!');
  });
});
