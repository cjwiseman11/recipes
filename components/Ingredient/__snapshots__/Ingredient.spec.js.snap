// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ingredient /> Ingredient is rendered with a span by default 1`] = `
<span
  class="ingredient "
>
  A tasty pie
</span>
`;
